cou/steps/__init__.py

```diff
@@ -51,29 +51,38 @@ def compare_step_coroutines(coro1: Optional[Coroutine], coro2: Optional[Coroutin
 
 
 class BaseStep:
-    """Represents each upgrade step.
+    """Represents a basic upgrade step.
 
-    This BaseStep is used to define any step when performing an OpenStack upgrade.
-    And requires description, parallel and coroutine as arguments. The coroutine is expected by
-    creating an asyncio.Task task instead of waiting directly, so that it is possible to cancel
-    the task and at the same time simply find out which task is really running according to
-    its name.
+    This BaseStep is used to as the bases for any step when performing an OpenStack upgrade.
+    It requires description, parallel, coroutine, and prompt as arguments. The coroutine is
+    expected by creating an asyncio.Task task instead of waiting directly, so that it is possible
+    to cancel the task and at the same time simply find out which task is really running according
+    to its name.
+
+    This class should not be used directly. Please use one of its child classes as fits.
     """
 
+    # pylint: disable=too-many-instance-attributes
+
+    prompt: bool = True  # whether to prompt for user input during execution
+
     def __init__(
         self,
         description: str = "",
         parallel: bool = False,
         coro: Optional[Coroutine] = None,
+        prompt: Optional[bool] = None,
```

Comment:
I do not understand what is benefit to have default value `None` and `True`  for class variable. Why using class variable and __init__ input variable? I think we should use only one and it should be class variable, since each type of step should behave same. e.g. `UpgradePlan` should not have prompt=False.
Maybe we will need to have some `UpgradeStep` which will have prompt=True, so in that class we may use prompt in as input variable for __init__ and change the default value, but this will be corner case situation which could be handled than.

Reply:
As per our sync outcome, I remove prompt from init function and only let it being controlled as an class attribute. In case in the future we need an `UpgradeStep` with prompt, we can create a separate class, e.g. `CriticalUpgradeStep` and set its `prompt` value to True.